File: app/cabinet-dentistry-backend/src/main/java/com/example/cabinetdentistrybackend/model/DoctorOwner.java
package com.example.cabinetdentistrybackend.model;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
@DiscriminatorValue("DOCTOR_OWNER")
public class DoctorOwner extends BaseUser {


    @Column(nullable = false)
    private String specialization;



    public DoctorOwner(String firstname, String lastname, String email, String password, RoleType roletype, String specialization) {

        super(firstname, lastname, email, password, roletype); // Appel du constructeur de la classe de base
        this.specialization = specialization;

    }









}

File: app/cabinet-dentistry-frontend/src/app/shared/models/DoctorOwner.ts
export interface DoctorOwner {
  id? : number;
  firstname : string;
  lastname : string;
  username : string;
  email : string;
  cin : string;
  phoneNumber : string;
  birthdate : Date;
  password : string;
  firstLogin : boolean;
  specialization : string;
  roletype : string;
}
